src/main/scala/nl.knaw.dans.easy.deposit/SubmitJob.scala

```diff
@@ -59,12 +62,16 @@ class SubmitJob(draftDepositId: UUID,
     submitDeposit() match {
       case Failure(e) =>
         logger.error(s"[$draftDepositId] error in dispatched submit action ${ this.toString }", e)
-        draftDepositStateManager.setStateFailed(e.getMessage)
-          .doIfFailure { case e => logger.error(s"[$draftDepositId] could not set state to FAILED after submission failed", e) }
+        draftDepositStateManager.setMailToDansDescription(State.submitted)
+          .doIfFailure { case e => logger.error(s"[$draftDepositId] could not set state description after submission failed", e) }
       case Success(()) =>
         sendEmail
           .doIfSuccess(_ => logger.info(s"[$draftDepositId] finished with dispatched submit action"))
-          .doIfFailure { case e => logger.error(s"[$draftDepositId] deposit submitted but could not send confirmation message", e) }
+          .doIfFailure { case e =>
+            // applying setMailToDansDescription (with something like no confirmation message sent)
+            // could get overwritten by easy-ingest-flow, or worse: overwrite a change by easy-ingest-flow.
```

Comment:
This is not true.

1. `draftDepositStateManager.setMailToDansDescription` is not supposed to be called here. The fact that sending an e-mail failed is no reason for the state to change.
2. the draft deposit is NEVER overwritten by `easy-ingest-flow` OR ANY OTHER APPLICATION. The draft deposits are managed by `easy-deposit-api` only. `easy-ingest-flow` has its own inbox. `easy-manage-deposit` updates the state of the deposit in `easy-ingest-flow` on publishing/rejecting a dataset, but it will NEVER touch the draft deposits in `easy-deposit-api`. The state of a draft deposit is only changed by `easy-deposit-api` (see `StateManager`).

I would suggest to remove this comment and restore the original code.

Reply:
`set...Description` does not change the state and it should not. The method that does change the state checks for a valid transition. 

It just changes the description when the background thread fails. No email is more or less a failure in the background thread, but in this case ingest-flow will start processing. "overwritten by" is a shortcut for "when the api peeks into ingest-flow to update the state".

I somehow was thinking whether we could tell the end user somehow that the confirmation mail was not sent.